commons-db/src/main/java/com/palantir/nexus/db/sql/AgnosticLightResultSetImpl.java

```diff
@@ -0,0 +1,233 @@
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.palantir.common.base.Throwables;
+import com.palantir.exception.PalantirInterruptedException;
+import com.palantir.exception.PalantirSqlException;
+import com.palantir.nexus.db.DBType;
+import com.palantir.nexus.db.ResourceCreationLocation;
+import com.palantir.nexus.db.monitoring.timer.SqlTimer;
+import com.palantir.nexus.db.sql.BasicSQLString.FinalSQLString;
+import com.palantir.sql.ResultSets;
+import com.palantir.util.AssertUtils;
+import com.palantir.util.Visitor;
+
+/**
+ * This result set only loads one row at a time, and thus provides a
+ * low-overhead solution for large queries.  Read the comments about the iterator before
+ * using it, because there are non-obvious pitfalls.
+ * @author dcohen
+ *
+ */
+class AgnosticLightResultSetImpl implements AgnosticLightResultSet {
+    private static final Logger log = LoggerFactory.getLogger(AgnosticLightResultSet.class);
+    private static final Logger sqlExceptionlog = LoggerFactory.getLogger("sqlException." + AgnosticLightResultSetImpl.class.getName());
+
+    private final ResultSet results;
+    private final DBType dbType;
+    private final Map<String, Integer> columnMap;
+    private final PreparedStatement stmt;
+    private final AgnosticLightResultRow singletonResultRow;  // This is here and not in the iterator so we can throw a SQLException when constructing it
+    private volatile boolean hasBeenClosed = false;
+    private SqlTimer.Handle timerKey;
+    private final String timingModule;
+    private final FinalSQLString sqlString;
+    private final ResourceCreationLocation creationLocation;
+    private final String creationThreadName;
+    private int maxFetchSize = DEFAULT_MAX_FETCH_SIZE;
+    private final SqlTimer sqlTimerFactory;
+
+    public AgnosticLightResultSetImpl(ResultSet rs, DBType type, ResultSetMetaData meta, PreparedStatement s,
+                                      String timingModule, FinalSQLString sqlString, SqlTimer sqlTimerFactory,
+                                      ResourceCreationLocation creationLocation) throws PalantirSqlException {
+        results = rs;
+        dbType = type;
+        stmt = s;
+
+        columnMap = ResultSets.buildJdbcColumnMap(meta, type);
+        singletonResultRow = new AgnosticLightResultRowImpl(dbType, columnMap, results);
+        creationThreadName = Thread.currentThread().getName();
+        this.creationLocation = creationLocation;
+
+        this.timingModule = timingModule;
+        this.sqlString = sqlString;
+        this.sqlTimerFactory = sqlTimerFactory;
+    }
+
+    @Override
+    public void close() {
+        timerKey.stop();
+        try {
+            stmt.close();
+            results.close();
+            hasBeenClosed = true;
+            long elapsed = System.currentTimeMillis() - creationLocation.getCreatingThreadInfo().getTimestamp();
+            log.info("Closed {} after {}ms", this, elapsed);
```

Comment:
do a pass through all of the code in this PR and redo the log levels such that an open source person logging at INFO+ won't see things that should essentially be debug or trace